### ISSUE TITLE
rogue behaviour on mate-desktop (marco WM +compositing)

### ISSUE THREAD

**Issue by pmp-p** [minetest/minetest#8841]
##### Minetest version 

```
Minetest 5.0.1 (Linux)
Using Irrlicht 1.8.3
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_GETTEXT=1
USE_SOUND=1
USE_CURL=1
USE_FREETYPE=1
USE_LUAJIT=0
STATIC_SHAREDIR="/usr/local/share/minetest"
```

##### OS / Hardware

Operating system: GNU/Linux  Mint 18/ubuntu xenial +  mate-desktop
CPU: amd a10-6800k

GPU model: amd a10-6800k
OpenGL version:

##### Summary

mouse + keyboard context still grabbed by minetest window when switching graphical app with Alt+Tab when not in game menu
( multimonitor setup )
##### Steps to reproduce

open mate-terminal + minetest 
run a game , do not go in game menu, press alt+tab try to type in terminal => no effect.

Possible bad XGrabPointer behaviour as discussed here https://bugs.freedesktop.org/show_bug.cgi?id=107358

**Comment by Wuzzy2** [comment]
I can confirm, this happened to me sometimes, too, but I don't know when exactly and did no deep research on it.

**Comment by Linuxdirk** [comment]
Can't confirm when started from urxvt terminal running in Openbox on plain X.

So maybe an DE or WM specific issue?

**Comment by pmp-p** [comment]
> So maybe an DE or WM specific issue?

my (very) wild first guess would be : let's find if irrlicht asks for fullscreen window behaviour via EWMH (ok) or  XGrabPointer (bad)

**Comment by pmp-p** [comment]
> So maybe an DE or WM specific issue?

my (very) wild first guess would be : let's find if irrlicht asks for fullscreen window behaviour via EWMH (ok) or  XGrabPointer (bad)

also noted if changing focus with an EWMH tool like xdotool like :
```sleep 4;xdotool keydown alt key Tab; xdotool keyup alt```
 the event blocking only happens for a few seconds, then windows swtiching happens  normally and terminal becomes active getting events  -> but game (is frozen not like when in menu)  and window behaves erratically when regaining focus which does *not* happends when in ingame menu.

**Comment by pmp-p** [comment]
according to @sfan5  EWMH events are directly handled by irrlicht. and as minetest window is properly responding to minimize/iconify commands i suspect a flaw regarding EMWH handling inside irrlicht 1.8.3 which does not cover all cases or mishandling them.

**Comment by style-nine** [comment]
Xfce does not exhibit this problem, but works as you would expect with Alt+Tab. I don't think this is related to X11 at all. I get normal behavior for both Gtk2 and Gtk3 apps running alongside Minetest.

**Comment by pmp-p** [comment]
@style-nine thanks for testing and reporting it helped me nailing down : 

it's a window manager compositing problem vs irrlicht ( up to 1.9 and GLES )  that affect "marco" the fork of metacity in mate desktop , and can be reproduced :
```marco --composite --replace```
-> launch MT and you have the very long switching delay
```marco --no-composite --replace```
-> no delay, expected sane behaviour

metacity is indeed not affected.

**Comment by An0n3m0us** [comment]
Can confirm here too on MATE 16.04 (marco software compositor). Not sure what causes it; but it's pretty easy to reproduce:
https://my.pcloud.com/publink/show?code=XZMuyskZbljTseXy06JNRyKYAHQb8b74geUk

I pressed Super+P on the keyboard to start the recording. Note where the cursor is; the issue only seems to occur when the cursor is close to the recorders window border.
At the end of the recording; when I tabbed into Firefox; one can clearly see that Minetest still has a hold of the cursor.
